protocol/x/subaccounts/keeper/isolated_subaccount.go

```diff
@@ -135,8 +186,148 @@ func isValidIsolatedPerpetualUpdates(
 	if isIsolatedSubaccount &&
 		hasIsolatedUpdate &&
 		isolatedPositionPerpetualId != isolatedUpdatePerpetualId {
-		return types.ViolatesIsolatedSubaccountConstraints, nil
+		return types.ViolatesIsolatedSubaccountConstraints, nil, nil
+	}
+
+	return types.Success,
+		getIsolatedPerpetualStateTransition(
+			settledUpdate.SettledSubaccount,
+			isolatedPerpetualPosition,
+			isolatedUpdate,
+		),
+		nil
+}
+
+// getIsolatedPerpetualStateTransition computes whether an isolated perpetual position will be
+// opened or closed for a subaccount given an isolated perpetual update for the subaccount.
+// Input subaccount account must be settled.
+func getIsolatedPerpetualStateTransition(
```

Comment:
For modularity, might make sense to have separate unit test for this function.

Reply:
Added separate unit tests.